### PR TITLE
Fix single quotes parsing (#125) plus test cases.

### PR DESCRIPTION
**Pull request by ghyatzo**
With this change, when we encounter a quote, we move through all elements within quotes, i.e `'abcde'`, leaving the past behaviour intact.
The added behaviour adds a check in the edge case where the quoted content begin with a quote itself, i.e `'''a'`.

In case we find another quote, right after the opening quote, we make sure that it is not the escaping single quote for a `'`
before quitting the while loop.

Initial tests are all green (see #125).
I added all the examples from the discussion as tests.

### CODE REVIEW

**Comment by ghyatzo** [comment]
I will first thing tomorrow morning!

**Comment by kescobo** [comment]
excellent! Thanks again - I just approved CI, once it passes I'll merge and tag a release.

**Review by kescobo — APPROVED** [review]

**Pull request merged**
